resolwe/flow/models/utils.py

```diff
@@ -339,45 +344,72 @@ def hydrate_input_uploads(input_, input_schema, hydrate_values=True):
                 value['file_temp'] = 'Invalid value for file_temp in DB'
 
 
-def hydrate_size(data):
+def hydrate_size(data, force=False):
     """Add file and dir sizes.
 
     Add sizes to ``basic:file:``, ``list:basic:file``, ``basic:dir:``
     and ``list:basic:dir:`` fields.
 
+    ``force`` parameter is used to recompute file sizes also on objects
+    that already have these values, e.g. in migrations.
     """
     from .data import Data  # prevent circular import
 
+    def get_dir_size(path):
+        """Get directory size."""
+        total_size = 0
+        for dirpath, _, filenames in os.walk(path):
+            for file_name in filenames:
+                file_path = os.path.join(dirpath, file_name)
+                if not os.path.isfile(file_path):  # Skip (broken) links
```

Comment:
It would be clearer to use [islink](https://docs.python.org/3/library/os.path.html#os.path.islink).

Reply:
Actually, i understood we wish to calculate sizes only for normal files. Any other structure (link is just an example that i bumped into) is potentially problematic. For example, links will point to certain location where the actual file is stored (and file size should be counted there, not here). Should i juts fix comment to make it more clear why file-like-ness is checked?

Reply:
> Should i juts fix comment to make it more clear why file-like-ness is checked?

Yes, thanks.